lib/active_model/serializer/configuration.rb

```diff
@@ -29,6 +29,28 @@ def config.array_serializer
         config.jsonapi_include_toplevel_object = false
         config.key_transform = nil
 
+        # example output
+        # => [
+        # "::Api::V1::ResNamespace::ResourceSerializer",
+        # "::Api::ResNamespace::ResourceSerializer",
+        # "::ResNamespace::ResourceSerializer" ,
+        # "::ResourceSerializer"]
+        config.serializer_lookup = Proc.new { |controller_name|
+          lookups = []
+          current_resource = controller_name.gsub(/Controller\z/, '')
+          current_resource += 'Serializer'
+          namespaces = current_resource.split('::')
+          resource_serializer = namespaces.slice!(-1)
+
+          lookups << '::' + resource_serializer
+          namespaces.length.times do |i|
+            current_namespace = namespaces[i - namespaces.length..-1].join('::')
+            lookups << '::' + current_namespace + '::' + resource_serializer
+          end
+
+          lookups
```

Comment:
oh!, @bf4 that serializer lookup defined in the controller wouldn't keep relationships under the same namespace, would it? I think that would be harder to debug than the proc here